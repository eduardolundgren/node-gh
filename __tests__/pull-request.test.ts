/**
 * Â© 2013 Liferay, Inc. <https://liferay.com> and Node GH contributors
 * (see file: CONTRIBUTORS)
 * SPDX-License-Identifier: BSD-3-Clause
 */

import { runCmd } from './testUtils'

describe('E2E: Pull Request Module Test', () => {
    it('List PRs `gh pr`', done => {
        expect(runCmd('bin/gh.js pr')).toMatchSnapshot()
        done()
    })

    it('Get PR Info `gh pr --info 50`', done => {
        expect(runCmd('bin/gh.js pr --info 50')).toMatchSnapshot()
        done()
    })

    it('Comment on PR `gh pr 50 --comment "Just started reviewing :)"`', done => {
        expect(runCmd('bin/gh.js pr 50 --comment "Just started reviewing :)"')).toMatchSnapshot()
        done()
    })

    it('Open PR `gh pr 50 --open`', done => {
        expect(runCmd('bin/gh.js pr 50 --open')).toMatchSnapshot()
        done()
    })

    it('Close PR `gh pr 50 --close`', done => {
        expect(runCmd('bin/gh.js pr 50 --close')).toMatchSnapshot()
        done()
    })

    it('Submit PR `gh pr -s protoEvangelion -b master -t "pr title" -D "pr description"`', done => {
        expect(
            runCmd('bin/gh.js pr -s protoEvangelion -b master -t "pr title" -D "pr description"')
        ).toMatchSnapshot()
        done()
    })
})
