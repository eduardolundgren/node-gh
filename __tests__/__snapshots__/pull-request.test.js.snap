// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E: Pull Request Module Test Close PR \`gh pr 50 --close\` 1`] = `
"Closing pull request #50
https://github.com/protoEvangelion/gh/pull/50
"
`;

exports[`E2E: Pull Request Module Test Comment on PR \`gh pr 50 --comment "Just started reviewing :)"\` 1`] = `
"Adding comment on pull request #50
https://github.com/protoEvangelion/gh/pull/50#issuecomment-000000001
"
`;

exports[`E2E: Pull Request Module Test Fetch PR \`gh pr --fetch 50\` 1`] = `
"Fetching pull request #50 into branch pr-50
[hook] gh pr 50 --user protoEvangelion --repo gh --comment 'Just started reviewing :)'
Adding comment on pull request #50
https://github.com/protoEvangelion/gh/pull/50#issuecomment-000000001
"
`;

exports[`E2E: Pull Request Module Test Get PR Info \`gh pr --info 50\` 1`] = `
"#50 pr2 @protoEvangelion (a year ago)
"
`;

exports[`E2E: Pull Request Module Test List PRs \`gh pr\` 1`] = `
"Listing open pull requests on protoEvangelion/gh
protoEvangelion/gh
master (3)
#51 pr3 @protoEvangelion (a year ago) ✓
#50 pr2 @protoEvangelion (a year ago) ✓
#33 test @protoEvangelion (a year ago) ✓
"
`;

exports[`E2E: Pull Request Module Test Open PR \`gh pr 50 --open\` 1`] = `
"Opening pull request #50
https://github.com/protoEvangelion/gh/pull/50
"
`;

exports[`E2E: Pull Request Module Test Submit PR \`gh pr -s protoEvangelion -b master -t "pr title" -D "pr description"\` 1`] = `
"Submitting pull request to @protoEvangelion
https://github.com/protoEvangelion/gh/pull/54
[hook] gh pr --browser  --user protoEvangelion --repo gh --number 54
"
`;
